### PR TITLE
Interdyne Overhaul

### PR DESCRIPTION
**Pull request by Jakksergal**
## About The Pull Request
Interdyne 2 in collaboration with ErdinyoBarboza's DS-3. THIS IS A STOP GAP INSTALLMENT.

Interdyne has been completely rebuilt from the ground up with a snazzy new facility intended to establish a more concrete identity in the server's canon. This is the first installment of a two-phase rebranding of Interdyne as a whole. As Interdyne is an entirely different company from NT I opted to give their home base a unique industrial aesthetic that will be improved upon over time. 

Chemistry has been expanded, giving them a medium-sized chemistry room to play around with and produce high volumes of chemiclal goods.

Interdyne's combat capabilities have been nerfed. Only the DO spawns with a sniper rifle currently and three bolt-action snipers are stored in the vault requiring a DO's access to open. Binoculars have been added to the gear crate to compensate for lost observation potential. Their offensive capabilities will be further reduced after DS-3 is released.
image

## How This Contributes To The Skyrat Roleplay Experience
Interdyne 2 is a much more efficient use of space compared to its predecessor and focuses purely on Interdyne's biggest export: medicine. This is phase one of a series of commits focusing on de-arming and establishing Interdyne as an independent faction to allow them to provide a better RP dynamic with the station than they currently do.

## Changelog
- Completely rebuilds Interdyne with a snazzy new facility focused on medical mechanics while temporarily retaining some of the old uses of the previous base.
- Interdyne's Xenobiology has been removed and the four Xenoarcheology machines have been placed outside in its stead.
- Interdyne scientists no longer spawn with .50 Cal. BMG auto sniper rifles. Three bolt actions have been added to a secure locker only the DO can access within the vault.
- All former access issues of Interdyne have been fixed.

:cl: Jakkie Sergal
add: Introduces Interdyne 2: A new facility for Interdyne agents.
balance: Interdyne scientists no longer spawn with automatic .50 cal snipers. DO can distribute 3 bolt action .50 cal snipers when needed.
fix: Fixes all access issues of the previous Interdyne base.
/:cl:

### CODE REVIEW

**Comment by softcerv** [comment]
could you include photos in your PR body?

**Review by Ebin-Halcyon — COMMENTED** [review]
A few missing things and a few things that aren't working correctly:

Missing the Syndicate mod suit storage, energy axe cabinet in engineering, the spare headsets and agent card crate, the Mining vendor isn't the Interdyne specific one TastyFish coded.
Didn't seem to find a button to unlock the front door to get out of the base, cremator doesn't have a button to turn it on. The positronic console broadcasts to the science channel so it's existence is moot for the Base.

**Comment by Jakksergal** [comment]
Fixed all above. Thanks map diff bot for dying with the last commit.

**Comment by theselfish** [comment]
Wow! So much space for three players!

**Comment by Ebin-Halcyon** [comment]
> Wow! So much space for three players!

Ghost cafe.

**Comment by Xymec** [comment]
I'll be honest here, Really bad idea, It's too easy to break in now. As for "RP and interacting with the station." We aren't supposed to be interacting with the station in general. And the .50 Cals were for keeping people away and killing fauna too close to interdyne. The Xenobio should stay considering there's no xenobio on DS-2 if DS-2 even spawned on that round, Xenoarcheology shouldn't be replacing it.

**Comment by Xymec** [comment]
The idea is a good concept but doesn't work well in game.

File: code/modules/mob_spawn/ghost_roles/mining_roles.dm [hunk 1]
```diff
@@ -303,7 +303,7 @@
 /datum/outfit/lavaland_syndicate
 	name = "Lavaland Syndicate Agent"
 	r_hand = /obj/item/gun/ballistic/automatic/sniper_rifle
-	uniform = /obj/item/clothing/under/syndicate
+//	uniform = /obj/item/clothing/under/syndicate // SKYRAT EDIT REMOVAL -- BALANCE
```
Comment (RatFromTheJungle) [inline code/modules/mob_spawn/ghost_roles/mining_roles.dm]: you uhh commented out their pants, not the gun

Comment (Jakksergal) [reply code/modules/mob_spawn/ghost_roles/mining_roles.dm]: Okay that's kinda funny. I'll fix it when Im next at my desk.

**Comment by softcerv** [comment]
Part of the appeal of Interdyne is that it serves as a more relaxed place to learn/teach game mechanics to newer players, without having a xenobio lab, you lose part of that. Removing the xenobio lab also prevents DS2 from doing xenobio too.

**Review by Jolly-66 — COMMENTED** [review]
I would like to review this when I'm not half asleep, but I did notice one thing off the bat.

You either need to make a custom, fire proof subtype that **cannot** be deconstructed, or find another way to add fluff here. The railings simple burn to rods and burn to ash over time.

<details>
<summary>Related image</summary>

image

</details>

I'll add onto this via another review when I'm not half asleep.

**Comment by Jakksergal** [comment]
@Xymec @softcerv On Xenobio:
Xenobio is only temporarily removed. Ganymede is working on adding DS-3 which will be a hard spawn using the centcom levels to ensure that it's loaded in every round, even on Blueshift or Icebox. It will have xenobio, toxins, and anything else that's normally gone when on the aforementioned maps. They're currently on hiatus and I wanted to get ahead of any issues my PR might have in it. I've added Xenoarch machines and will give equipment in the next PR after Jolly does their review.

On guns: You're right, Interdyine's not supposed to be interacting with the station but this de-escalation of arms is a step towards policy change. A full Interdyne crew with DS-2 present can sweep the floor with the entire station with those 50 cals if they have marksmans of sopo rounds from DS-2's ammo printer. Interdyne still gets .50 cals but they're in a locker only the DO or higher can unlock, which falls in line with the current standing policy of needing the DO's permission to piss off miners.

**Comment by RatFromTheJungle** [comment]
swap the .50s in the armory for m16/oldarm's, or maybe cantalan stuff (wildcats, lyxes single spare magazine per gun), or, thirdly, shotguns of a variety, with a spread of shell-boxes, for-when they start going stationside, most of their weapons need to be generally stuff already available to the crew, (and also to prevent them immediately giving .50s to sec or their friends or w/e)

elsewise, maybe some weaker (energy?) turrets spaced around on the lava, to discourage people just immediately RCD'ing over to them (while the guns are active), nothing insta-deadly like the main-bases guns

and for a final one; maybe replace the emag in the vault with an ammo-bench board, and chuck in a few more pistol mags until DS3 is around.

**Comment by Jakksergal** [comment]
Personally I'm waiting for an alternative to MCR rifles that will hopefully be available by the time DS-3 comes along to replace the .50s with. They are bonkers strong but I tested their fire rate a few hours ago and they're much, much less effective than the ones they previously spawned in with. Trying to de-escalate in baby steps to see how players feel about the changes. 

I could add a few turrets here and there but their AI isn't the best and I can't use energy, as they'd be totally ineffective to fauna. I'm not too concerned about people RCDing in and stealing everything because, frankly, that's something that should be opfor'd and was totally possible with the previous Interdyne. Also I didn't mention it but I removed all instances of one tile (orthogonally) of lava separating the base from mainland.

**Comment by Whygil** [comment]
I think this is too much space for 7 people. Current Interdyne already takes a good chunck of Lavaland, not sure of taking more space is a good idea

**Comment by Kush1Push1** [comment]
I can land a plane in this hallways
Lavaland seems to be tight after adding such large ruins

<details>
So much free space...

opera_vpmfvl0hBv

opera_I1b0na5mSQ

opera_ZpmJqqtgfX

Spaaace

opera_uKf5HFf4p7

You have DS-2 for chem factories, there is much more space for them

opera_yqByxLw5R2

A bar, a cafe, a kitchen... 

opera_RFM6GE3COx

</details>

I would have vibes of a ghost ship/colony in a place like this

**Comment by Kush1Push1** [comment]
> > Wow! So much space for three players!
> 
> Ghost cafe.

It doesn't wastes general playing areas, like station or lavaland.

**Comment by Jakksergal** [comment]
@Whygil @Kush1Push1 The X/Y footprint for this is only larger by 4 tiles in one axis, making it roughly the same size as the existing Interdyne base. If you wanna talk about wasted space, make a issue report about the zig-zag halls and nothing ruins added. This is intended for players to actually use and get something out of. 

Even with the wasted space, this uses it far more efficiently than the existing interdyne which is just a big square of lava with a plus shaped building in the middle of it.

Bottom left is landing area for a shuttle which will replace the telepad when DS-3 arrives. Bottom right is for a chem lab, as DS-3 will not have one. Bottom middle will be a construction area just for funzies if the Interdyne crew want to make a pool or something. It's better than just more lava. It's currently a Syndie Sleeper Deluxe+ with R&D facilities for the sake of Icebox and Blueshift.

**Review by Jolly-66 — CHANGES_REQUESTED** [review]
Alright. I'm on my feet, here we go.

### Camera Dead Zones
You lack cameras in these locations, or do not have sufficient coverage:

<details>
<summary>Camera Dead Zones</summary>
Right here in the laundry room.

image

Over here in medical, there's patches of dead zones.

image

Morgue.

image

Very awkward dead zone outside the bar.

image

Ditto.

image

The entrance to the place.

image

</details>

### Access Helpers

This is not how you use the access helpers. They're only for airlocks. Set their req_access with `list("access_type")`.

<details>
<summary>Windoors</summary>
Monkey pen.

image

Bar.

image

</details>

### Misc Stuff
Here's some other noteworthy things.

<details>
<summary>Images</summary>

The lattice here make no sense. Remove them.

image


The quantum pad can be moved to the shuttle for some RP flavor of "I'm taking the shuttle".

image

The dorms shouldn't be controlled by one lockdown button. Add one for each cabin. The button for the lockdown is also in an awkward spot in general, as someone would assume it would close the hall, not open the window shutters.

image


Move the LustWish vendor to the bathroom.

image

Change the airlocks of the Deck Officer to QM doors.

image

Please change the grass turfs used in Hydroponics. Use another tile turf and add trim lines. Grass doesn't really work well with the environment its in. If this were a proper bee room, I wouldn't mind, but it needs to be reworked. Also scatter the items on the rack a bit, it feels bloated.

image

This area lacks lighting near the top most tables.

image

Replace the rods with sheets. This is just standard and shouldn't be an issue.

image

Add a morgue door here, and move the NanoMed down. This way Syndicate Doctors don't need to haul bodies off through the hall.

image

The thermomachine should be set up.

image

We don't put Booz-o-Mats in walls anymore. Rearrange how you have things set up.

image

Please rename this cabinet. When I was testing I thought there was going to be a **regular** energy axe.

image

You don't need just plating here. Pipes can be freely displayed on top of tiles, making this look incredibly jank.

image

I'm not a fan of how disposals is being done. Either have it be proper (like a disposals room that then shoots out into lava) or have it dump directly into lava.

image

Please put a window here, pipes under walls is incredible sin.

image

image

</details>

My overall and two final critiques are these,

- Shrink the entire areas halls down by one

Theres ~7 spawners, and too much space. Shrink ALL the halls down by one tile so they're not 3x3, instead, 2x2. You get the idea.
- Change the atmos pipes back to red/blue

I know you were passionate on shaking things up, but I was confused initially. Change the grey/orange back to blue/red for readability. I should mention, if we have any colorblind players, both now or in the future, who decide to play this, will have a much harder time adapting.

**Comment by Jakksergal** [comment]
@Jolly-66 
> Camera coverage

Not all areas need to be covered just as they aren't on the main station. Why does it matter if the laundry room doesn't get coverage?
Will fix others.

> Lattices and quantum pad

They're visual fluff. That's where the shuttle lands when DS-3 arrives. The airlock is temporarily being used for the DS-2 pad. The lab is also temporary. (Also it's a Syndie Sleeper camper van.)

> Lockdown button and cabins

The lockdown is mostly to prevent mega fauna from aggroing to people inside the base and function as a 'porch light' to show miners that Interdyne players are awake. Intended to put curtains in each dorm.

> Lustwish

.. Why? Does it really matter?

> Doors

This is super nitpickey.. Changed the shade of brown on the front door.

> Grass

It's flavor. They're fake grass turfs created in botany.

> Low lighting

Atmosphere. Lava provides light, candles provide light. I'll add one more bulb so it's at least not pitch black at the topmost part.

> Rods

Rods have 50% less material value per unit so they are fill the plumbing tool closer to its cap when partially emptied. It is the same total value as one stack of iron.

> Morgue/Surgery

No. It's cramped enough as it is. They didn't even need a morgue in the first place.

> Cryo

I don't want everything to be setup and handed to Interdyne. They already get T4 parts for everything, who cares if they have to click 5 times and wait a little bit to get their magic heal juice tube going. Might even move the tank to engineering so it's a little routine.

> Bar

Oh you're gonna hate me. The booze-o-mat is on the tile of the stool and shifted upwards so it's not in the wall. Yes it's set to open so it can be walked 'on'. No I'm not moving the fridge. It's blocked on all orthagonal faces to the outside.

> Energy axe

What? Make up your mind, do I var edit or not? This seems pointless. And it is a regular energy axe.

> Engineering

Mimics the style of the current interdyne. Forgot to add hazard stripes, though.

> Disposals

Huh? It is a disposal 'room'. You have a chance to recover anything you didn't want in the trash and can burn the rest. Short of a mass driver, complicated blast door controller, and belt spaghetti it's still a disposal area.

> Pipes under walls

It's 3 tiles. Actually I'm gonna tweak viro so that it snakes under the same tile as the scrubber line to save on a tile definition.

> Shrink the halls

This does nothing to shrink the footprint of the base. Rule of 3rds, shrinking the halls to be half grey half black ruins the aesthetic I was going for originally. Not to mention it's a LOT of work for zero gain.

> Pipe colors

Pardon but you're only one person. It's a simple enough change if it hits live.

I will note though that the white-orange coloration here is ***EASIER*** to see than red/blue in all conditions except for on top of white tiles and that's for all ranges of color blindness with the exclusion of the 0.003% of the population that has monochromatic color blindness. Even then, however, it just appears as a lighter version of the standard color codes.

**Comment by GoldenAlpharex** [comment]
> > Pipe colors
> 
> Pardon but you're only one person. It's a simple enough change if it hits live.

Good sir, they might be one person, but they're also the one person that's likely going to be in charge of merging your PR.

Simple enough or not, it's confusing for no valid reason, and goes against how it's done literally *everywhere* else.

**Comment by Xymec** [comment]
> Personally I'm waiting for an alternative to MCR rifles that will hopefully be available by the time DS-3 comes along to replace the .50s with. They are bonkers strong but I tested their fire rate a few hours ago and they're much, much less effective than the ones they previously spawned in with. Trying to de-escalate in baby steps to see how players feel about the changes.
> 
> I could add a few turrets here and there but their AI isn't the best and I can't use energy, as they'd be totally ineffective to fauna. I'm not too concerned about people RCDing in and stealing everything because, frankly, that's something that should be opfor'd and was totally possible with the previous Interdyne. Also I didn't mention it but I removed all instances of one tile (orthogonally) of lava separating the base from mainland.

There is no points at where you could be shot at so RCDing into the base is 75% easier now, lockdown will do jackshit against megafauna unless you make the airlocks or well, the whole base itself indestructible, with guncargo now around, if people OpFor breaking into interdyne, killing people there with obviously mechanically unfair weapons is very likely to happen as well, meaning now that the people there will be unarmed until a DO or someone on DS-2's higher ranks spawn, alongside that, sec raids, there should be enough weapons to hold off 8 officers, cause they dominate interdyne in terms of equipment, especially with the fact they get .45s in their armory. Weapons shouldn't be locked until a certain role spawns cause it's like locking TC behind submitting an OpFor, it's just bad. In my opinion, the interdyne we have right now doesn't need any changes, it's good as it is and works well for everyone.

**Comment by Xymec** [comment]
also, the sniper rifles also kind of were for shooting people fucking with the base or fauna that are getting too close, example: take a look at the ice watcher on ice box, it fucks over interdyne to a point of where you WILL need something that fucks them over. And they're syndicate, ballistics is kind of their whole thing, MCR rifles won't exactly make sense on syndicate roles cause the syndicate is based around using ballistic weaponry.

**Comment by StrangeWeirdKitten** [comment]
Alright, I do play a fairly decent amount of Interdyne/DS-2 and I'd like to go over my opinion and the overall points people are throwing out.

Overall, I like the general layout of the map and scientists + DO spawn with bolt action .50 cals currently. Does Interdyne need that many snipers? Not really. The snipers are awful for fauna, **especially megafauna**, regardless and you'd be better off getting a PKA from mining equipment and shooting them with that. They're mostly there as a deterant for players to fuck with Interdyne. I don't support the complete disarmament of Interdyne because the whole theme around Interdyne, from what I know currently, is that it's the secret underground syndicate branch of the company doing syndicate things. They are not someone the average shaft miner should fuck with. I can count on one hand the amount of times someone truely messed with Interdyne and one .50 cal with lethals and a pistol was plenty. Security raids don't happen to Interdyne even when shit gets heated because security is always busy with their own crap. The station straight up tells the miners not to fuck with the lava base and that if they do, they're on their own. Plus, they have omni-lathes and the ability to make combat mechs. 

> Interdyne scientists no longer spawn with .50 Cal. BMG auto sniper rifles

They already don't? It's just .50 cal bolt actions with one clip of lethal ammo for each.

> The map is too big and it's wasting "space"

Interdyne has for ages always had a fairly consistant population that will exceed 10+ people including with the people who spawn from DS-2. I don't know why space overall is being put up as a big deal. The base looks bigger, sure. But the foot print is not any bigger. Mapp Diff Bot outlines it pretty well. I think the 3x3 halls look fine. 

Like Jakk said, if you want to look at wasted lavaland space, take a look at the nothing ruins like the massive, indestructable zig zag slow floor thing which only rewards you with a mutated orange at the end of it.

**Map Diff Bot**
https://cdn.discordapp.com/attachments/948309285199310929/989340505680781352/unknown.png

> DS-3 on Central Command

I really don't think this is a good idea. Central Command levels are more designed as a launching off stage for antags. Not for full playable possitions unless you make the walls completely indestructable, which I do not support. You don't want the syndicate dudes space tiding over central command and this shouldn't merely be chalked under the rug under the idea of "Admins can handle that" I think how the current DS-2 spawns is fine. 

> Xenobiology

Just put a xenobiology under where the quantam pad is. That's another mechanical thing people enjoy to mess with and there is no reason to completely strip it away. I'd rather have a functional xenobiology then an asthetic bit of empty space.

> The Lustwish

This seems really nit pickey and honestly, most people can handle the sight of a purple heart kinky vend. It's nothing diffrent then the lustwish being fully visible in places like the locker room and dorms on the main station.

>Dispolsals

Perhaps a mass driver would benifit here with something to "catch" the trash and put it on the lava. Some shit can explode when they burn. 

> Pipe Colors

I don't think we need to break off from the standard red + blue. 

Honestly, at the end of the day. I feel that DS-2 is more fluff and Interdyne is overall generally more the place people tend to hang out. Half of DS-2 is maintenance, a pool, and other ammenaties just to add more bulk to the station for the sake of bulk. For when it comes to DS-3, I'd love to see an atmospherics and some stuff it doesn't currently have and this'll make up for moving xenobiology to DS-3.

**Comment by Jakksergal** [comment]
After consideration of Interdyne's OOC role in letting both experienced players and new players experiment and dick around, I'm convinced removing xenobio and shoe-horning in Sci isn't in good faith to the development process. I was just eager to get something published.

Closing PR until DS-3 is closer to release.

**Pull request closed without merging**